app.go

```diff
@@ -244,24 +246,31 @@ func pushImage(image string) {
 	execBuild("./", []string{"docker", "push", image})
 }
 
-func buildImage(image string, handler string, functionName string, language string, nocache bool) {
+func buildImage(image string, handler string, functionName string, language string, nocache bool, squash bool) {
 
 	switch language {
 	case "node", "python":
 		tempPath := createBuildTemplate(functionName, handler, language)
 
 		fmt.Printf("Building: %s with Docker. Please wait..\n", image)
 
-		cacheFlag := ""
+		//lets create a buffer for any build flags
+		var flagBuf bytes.Buffer
+
+		// test which build flags are required and add them to the buffer
 		if nocache {
-			cacheFlag = " --no-cache"
+			flagBuf.WriteString("--no-cache ")
+		}
+		if squash {
+			flagBuf.WriteString("--squash ")
 		}
 
-		builder := strings.Split(fmt.Sprintf("docker build %s-t %s .", cacheFlag, image), " ")
-		if len(os.Getenv("http_proxy")) > 0 || len(os.Getenv("http_proxy")) > 0 {
-			builder = strings.Split(fmt.Sprintf("docker build %s--build-arg http_proxy=%s --build-arg https_proxy=%s -t %s .", cacheFlag, os.Getenv("http_proxy"), os.Getenv("https_proxy"), image), " ")
+		if len(os.Getenv("http_proxy")) > 0 || len(os.Getenv("https_proxy")) > 0 {
+			flagBuf.WriteString("--build-arg http_proxy=" + os.Getenv("http_proxy") + " ")
```

Comment:
Should this be changed to sprintf since we are concatenating strings here and in other places use `fmt.Sprintf`?